sw/device/lib/base/status.h

```diff
@@ -95,12 +95,32 @@ typedef struct status {
        dif_result_t: DIF_RESULT_INTO_STATUS(expr_))
 // clang-format on
 
+/**
+ * Report an error status. This header does not specify how the error is
+ * reported, or if it reported at all. Since status_t encodes the location
+ * of an error, this provides the infrastructure for a lightweight
+ * "stack trace". */
```

Comment:
nit: format for these doxygen comments are similar to commit messages:
```suggestion
 * Reports an error status. 
 *
 * This header does not specify how the error is
 * reported, or if it is reported at all. Since status_t encodes the location
 * of an error, this provides the infrastructure for a lightweight
 * "stack trace". */
```

Applies throughout.